src/server/models/EventVolunteer.ts

```diff
@@ -0,0 +1,38 @@
+import { EventVolunteerEntity } from '@/types/dataModel/eventVolunteer';
+import { Model, Schema, model, models } from 'mongoose';
+
+const EventVolunteerSchema = new Schema(
+  {
+    role: {
+      type: Schema.Types.ObjectId,
```

Comment:
this should just be an enum, check the `Volunteer` model and look at it's `previousRole` field